tests/integration/models/test_evaluate_constitutive_laws.py

```diff
@@ -0,0 +1,252 @@
+            None,
+        ),
+        (
+            _add_mixin(pp.constitutive_laws.BartonBandis, setup_utils.MomentumBalance),
+            "elastic_normal_fracture_deformation",
+            # (-normal_traction) * maximum_closure /
+            #    (normal_stiffness * maximum_closure + (-normal_traction))
+            (1 * 0.1683) / (1529 * 0.1683 + 1),
+            1,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            1e-20,
+            2,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            0.01**3 / 12,
+            1,
+        ),
+        (
+            _add_mixin(c_l.CubicLawPermeability, setup_utils.MassBalance),
+            "permeability",
+            0.01**4 / 12,
+            0,
+        ),
+    ],
+)
+def test_evaluated_values(model, method_name, expected, dimension):
+    """Test that the value of the parsed operator is as expected.
+
+    Parameters:
+        model: Model class to be tested. Likely composed of several mixins.
+        method_name: The specific method to be tested.
+        expected: The expected value from the evaluation.
+        dimension: Dimension on which the method will be invoked. If None, all
+            dimensions in the MixedDimensionalGrid will be considered.
+
+    """
+    # The thermoporoelastic model covers most constitutive laws, so we use it for the
+    # test.
+    # Assign non-trivial values to the parameters to avoid masking errors.
+    solid = pp.SolidConstants(setup_utils.granite_values)
+    fluid = pp.FluidConstants(setup_utils.water_values)
+    params = {
+        "material_constants": {"solid": solid, "fluid": fluid},
+        "fracture_indices": [],
+    }
+
+    setup = model(params)
+    setup.prepare_simulation()
+
+    # Set variable values different from default zeros in iterate.
+    # This yields non-zero perturbations.
+    if hasattr(setup, "pressure_variable"):
+        setup.equation_system.set_variable_values(
+            2 * np.ones(setup.mdg.num_subdomain_cells()),
+            [setup.pressure_variable],
+            to_iterate=True,
+        )
+    if hasattr(setup, "temperature_variable"):
+        setup.equation_system.set_variable_values(
+            3 * np.ones(setup.mdg.num_subdomain_cells()),
+            [setup.temperature_variable],
+            to_iterate=True,
+        )
+    method = getattr(setup, method_name)
+
+    # Call the method with the domain as argument. An error here will indicate that
+    # something is wrong with the way the method combines terms and factors (e.g., grids
+    # parameters, variables, other methods) to form an Ad operator object.
+    # TODO: At some point, we should also consider interface laws.
+    op = method(setup.mdg.subdomains(dim=dimension))
+    if isinstance(op, np.ndarray):
+        # E.g. permeability or conductivity
```

Comment:
I thought permeability and conductivity return operators?

Reply:
Indeed, rewrote the comment. I suggest we still keep the option, although I'm not sure it ever kicks in.